### DISCUSSION TITLE
define a global variable that needs to be updated after every substep

### DISCUSSION THREAD

**Discussion by eric199405** [idaholab/moose#24995] [Q&A Modules: Solid mechanics]
Hello everyone. I am using MOOSE to run the CPFE simulation. I have a global variable which needs to loop every _qp to obtain. Besides, this variable needs to be updated every substep. I try to implement it under Materials, but fail. 

How can I realize this function? I think I should add an Userobjects in my APP. Would you please simply write down the loop for each _qp and fine the maximum _pk2?

**Comment by GiudGiud** [comment]
Hello

this seems like the behavior for a postprocessor
If you write down the behavior that should happen at every quadrature point you can use the postprocessor loop to get it to do what you want

Guillaume
